### PR TITLE
Fix record_type issue

### PR DESCRIPTION
**Pull request by Mobius1D**
If the transformation function `f` changes returns a different type then the it can't be converted to example type as the code would do before. So it is better to keep the type similar based on whether f is a Type or not. 

This feature is highly useful in RLDatasets where we would have to convert the type of Example into our required type.

### CODE REVIEW

File: src/core.jl [hunk 1]
```diff
@@ -42,26 +42,27 @@ function read_record(io::IO)
...
 - `channel_size=1000`. The number of pre-fetched elements.
-- `eltype=Example`. Change it to the type of result `f(::Example)` if `f` is provided.
+- `record_type=Example`. The type of value being read from the `file/files` that are provided.
 
 !!! note
 
     To enable reading records from multiple files concurrently, remember to set the number of threads correctly (See [JULIA_NUM_THREADS](https://docs.julialang.org/en/v1/manual/environment-variables/#JULIA_NUM_THREADS)).
 """
-read(s; kw...) = read(identity, s; kw...)
+read(s; kw...) = read(s; kw...)
```
Comment (findmyway) [inline src/core.jl]: I think you might have some misunderstanding here.

The data type read from tensorflow record is always `Example`. So we don't need to specify it explicitly. Here the `record_type` means the datatype of the result `f(x::Example)`.

File: src/core.jl [hunk 2]
```diff
@@ -42,26 +42,27 @@ function read_record(io::IO)
...
 Read tensorflow records from file(s).
 
 # Keyword Arguments
 
+- `f=Example`. Changes the type of the elements in the channel to the type specified by `f` if provided.
 - `compression=nothing`. No compression by default. Optional values are `:zlib` and `:gzip`.
 - `bufsize=10*1024*1024`. Set the buffer size of internal `BufferedOutputStream`. The default value is `10M`. Suggested value is between `1M`~`100M`.
 - `channel_size=1000`. The number of pre-fetched elements.
-- `eltype=Example`. Change it to the type of result `f(::Example)` if `f` is provided.
+- `record_type=Example`. The type of value being read from the `file/files` that are provided.
```
Comment (findmyway) [inline src/core.jl]: I think only this line needs to be updated.

File: src/core.jl (hunk 1 above)
Comment (Mobius1D) [reply src/core.jl]: Ahh, I see.

File: src/core.jl (hunk 2 above)
Comment (Mobius1D) [reply src/core.jl]: I will make the necessary changes right now

**Comment by lassepe** [comment]
I have trouble understanding the PR description; could you please elaborate why this breaking change was necessary? Why do you only want to support the `Example` record type? In my understanding tfrecord is not a format limited to the example protos and outside of ML there are good reasons to have other record types (which I use this package for and which was the reason why I introduced the `record_type` in #11).

**Comment by findmyway** [comment]
@Mobius1D

**Pull request merged**